ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java

```diff
@@ -196,6 +203,13 @@ public StatementExecutorResponse execute(
           statement.getStatementText()), e);
     }
 
+    if (!rateLimiter.tryAcquire()) {
```

Comment:
I think it would make sense to tryAcquire with a delay of a second here. This way we don't fail scripts that serially try to enqueue a large number of commands.

Reply:
@rodesai just added -> will you confirm this is what you were thinking?